src/extensions/default/BrambleUrlCodeHints/camera/interface.js

```diff
@@ -101,6 +117,98 @@ define(function (require, exports, module) {
         // Update the photo component with the snapped photo
         this.photo.update();
 
+        // pattern from http://codepen.io/SitePoint/full/LVpNjp/
+        // $(function() {
+
+          //var Caman = require('caman').Caman;
+          var canvas = document.getElementById("canvas");  // canvas reference here
```

Comment:
@humphd I've managed to successfully implement all your other suggestions, but this one results in a weird bug.  Apparently the image is taken, but the Caman canvas never gets loaded. I'll see if I can't debug this tonight.